include/SH3/camera/camera.hpp

```diff
@@ -0,0 +1,169 @@
+/** @file
+ *
+ *  OpenGL Version of the SILENT HILL style camera.
+ *
+ *  The camera works on a few basic principles, firstly, we consider that Heather (or James),
+ *  is the origin that the camera is <i>always</i> looking at. The camera can be in it's own position,
+ *  however for the most part (not including cutscenes), we are always looking at the protagonist.
+ *
+ *  Next, the camera seems to be in its own 'volume' (for lack of a better term), in that it <i>can't</i>
+ *  under any circumstances clip through a wall. There's a few links (albeit for different game engines) that
+ *  suggest how it is possible to do this, found here:
+ *  <a>https://www.reddit.com/r/Unity3D/comments/38qsx1/camera_collision_silent_hill_style/?st=j71bhmzc&sh=28a00dc6</a>
+ *  <a>https://answers.unrealengine.com/questions/169933/set-up-a-camera-like-silent-hill-and-resident-evil.html</a>
+ *
+ *  I googled <a href = https://www.google.com.au/search?q=silent+hill+style+camera&oq=silent+hill+style+camera&aqs=chrome..69i57j69i60j69i57j69i65j69i60l2.2672j0j7&sourceid=chrome&ie=UTF-8>this</a> term.
+ *
+ *  It would seem that camera data is stored in .cam files in the arc sections. I assume that this data is fed into the camera
+ *  when some kind of trigger (an AABB??) is triggered when Heather walks over/through it. It is currently
+ *  unknown where this data is stored, however, Mike seems to think it is in the .cld files (Collision & Lighting Data [CLD]).
+ *
+ *
+ *  There are a few modes the camera can be in:
+ *      STATIC - Static camera view, usually in tight spots (the bookstore store room).
+ *      FOLLOW - Follows Heather around (outside areas).
+ *  @copyright 2017 Palm Studios
+ *
+ *  @date 1-9-2017
+ *
+ *  @author Jesse Buhagiar
+ */
+#ifndef CAMERA_HPP_INCLUDED
+#define CAMERA_HPP_INCLUDED
+
+#include "glm/glm.hpp"
+
+namespace sh3{namespace camera{
+
+    /**
+     *  Camera Modes.
+     */
+    enum class CAMERA_MODE
+    {
+        FOLLOW,
```

Comment:
As none of these modes are implemented yet, they shouldn't have associated values. Comment them out.